src/v/security/license.h

```diff
@@ -62,7 +62,8 @@ inline std::ostream& operator<<(std::ostream& os, license_type lt) {
     return os;
 }
 
-struct license : serde::envelope<license, serde::version<1>> {
+struct license
+  : serde::envelope<license, serde::version<1>, serde::compat_version<0>> {
```

Comment:
this is a cool fix. 

@jcsp , @dotnwat and @mmaslankaprv  - should we make `compat_version<>` explicit. No implicit versioning, in fact code should fail without compat-version? 

maybe we check in with @felixguendling  - seems like the issue was it compiled without serde::compat*

Reply:
the issue was that the compat version had the wrong default. i think it is now explicit, iiuc.